### DISCUSSION TITLE
Does aiohttp support proxy protocol?

### DISCUSSION THREAD

**Discussion by mineus55** [aio-libs/aiohttp#7948] [Q&A]
I've been using nginx reverse proxy that redirects a domain to upstream web servers including aiohttp. There is this feature called proxy_protocol that broke request to aiohttp when enabled. Is it supported?

**Comment by Dreamsorcerer** [comment]
No, it looks like something that essentially wraps HTTP (i.e. it adds data _before_ the HTTP message). There would need to be changes to the parsers etc. to be able to handle this, if I understand correctly (and it was actually something we wanted to support). I'm not clear what advantage this would add to aiohttp though, at a glance it doesn't look useful for a web application (seems to be more useful for VPN, firewalls or similar when bounced through a proxy).

**Reply by webknjaz** [reply]
Years ago, the Home Assistant folks tried hiring me to implement something like that. And I even looked into where it could be integrated. But then, we lost contact (I stopped getting email replies) so I didn't proceed any further. I think their use case was a reverse proxy too.
